### PR TITLE
Success stories pages (Sprint 12)

### PR DESCRIPTION
**Pull request by Tehsurfer**
# Description

Defined for sprint 12 by the following tickets:
https://www.wrike.com/open.htm?id=660279044
https://www.wrike.com/open.htm?id=660277446
https://www.wrike.com/open.htm?id=660274680

Changes can be viewed on a sparc-app clone at:
https://sparc-succcess-stories.herokuapp.com/

Designs can be viewed at:
https://share.goabstract.com/fd46a371-7a46-4f30-a342-3823018d50a4

### Changes

- Added 'news-and-events/community-spotlight' page
- Added 'news-and-events/community-spotlight/success-stories' page
- Added page for each success story at 'news-and-events/community-spotlight/success-stories/<success-story>'
- Added rich text processing for contentful (this was required as far as I could tell to embed videos in text)
- Updated 'news-events' page with community spotlight section
- Created a 'Dataset Card' component for displaying dataset info in small areas

#### Updated news and events page
image

#### Success story details page
image

#### Success Stories landing page
image

## Type of change

Delete those that don't apply.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Run locally and tested at:
https://sparc-succcess-stories.herokuapp.com/

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by alan-wu** [comment]
@Tehsurfer I had a quick look at this [page](https://sparc-succcess-stories.herokuapp.com/news-and-events/community-spotlight/success-stories/delivery-of-electrical-signals%20to-vagus-nerve) and notice one of the words in the last assoicated dataset item is too for the container. Adding `word-wrap: break-word; ` to the dataset-description class may help.
It will take me a while to go through the codes.

**Comment by Tehsurfer** [comment]
> word-wrap: break-word;

Thanks Alan this works.

> The iframe at the bottom of this page is not rendered correctly

Sorry, iframe was actually plain text I had in contenful from testing

File: pages/news-and-events/community-spotlight/success-stories/index.vue [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      <p>
+        Learn how researchers have used the SPARC Program to advance
+        neuromodulation of the ANS. Submit your success story
+      </p>
+    </page-hero>
+    <div class="page-wrap container">
+      <community-spotlight-listings :stories="shownStories" />
+    </div>
+    <pagination
+      v-if="allStories.length > pageSize"
```
Comment (alan-wu) [inline pages/news-and-events/community-spotlight/success-stories/index.vue:16]: Typo?

File: pages/news-and-events/community-spotlight/index.vue [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      <p>
+        Hear about how the SPARC community are using the SPARC portal, including
+        Success Stories and Fireside Chats. Also, keep up to date with Community Announcements.
+      </p>
+    </page-hero>
+    <div class="page-wrap container">
+      <community-spotlight-listings :stories="shownStories" />
+    </div>
+    <pagination
+      v-if="allStories.length > pageSize"
```
Comment (alan-wu) [inline pages/news-and-events/community-spotlight/index.vue:16]: Typo?

Comment (Tehsurfer) [reply pages/news-and-events/community-spotlight/index.vue:16]: for `pagination`? That's how the component is spelled in sparc-app.

https://github.com/nih-sparc/sparc-app/blob/master/components/Pagination/Pagination.vue

File: components/CommunitySpotlight/CommunitySpotlightListings.vue [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      default: false
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import "@/assets/_variables.scss";
+
+.seperator-path {
```
Comment (egauzens) [inline components/CommunitySpotlight/CommunitySpotlightListings.vue:44]: Not sure if this is desired or not. Looks a little weird in a full screen browser that the separator line doesn't match the width of the text. 
image

File: pages/news-and-events/community-spotlight/success-stories/_id.vue [hunk 4]
```diff
@@ -0,0 +1,333 @@
+<template>
```
Comment (egauzens) [inline pages/news-and-events/community-spotlight/success-stories/_id.vue:1]: Wireframes show a link to the success stories page under the details: 
image

**Review by egauzens — CHANGES_REQUESTED** [review]
Overall changes look good! Just a couple comments to address about matching the wireframes. 

The only other real concern I see with the changes are for smaller screens. These changes do not take into account mobile. If you view the webpage on a smart phone you will see what I mean. I would try to follow the same layout pattern as the other components on the page since the original design seems mobile friendly

File: pages/news-and-events/community-spotlight/success-stories/_id.vue (hunk 4 above)
Comment (Tehsurfer) [reply pages/news-and-events/community-spotlight/success-stories/_id.vue:1]: Thanks I missed that

File: components/CommunitySpotlight/CommunitySpotlightListings.vue (hunk 3 above)
Comment (Tehsurfer) [reply components/CommunitySpotlight/CommunitySpotlightListings.vue:44]: Yeah the separator is fixed width will fix this

**Comment by Tehsurfer** [comment]
@egauzens Thanks for the review!

Sounds good, I'll use some of the styling in the news/events page for how to display these pages on mobile

**Comment by Tehsurfer** [comment]
@egauzens  Page have been updated to display nicely on mobile. Let me know if that meets the changes you were thinking of

**Review by egauzens — APPROVED** [review]
Changes look good, nice job!

**Comment by Tehsurfer** [comment]
Merging this in so it's ready for to be sent out for the upcoming portal CCB meeting

**Pull request merged**